_drafts/2020-09-17-fixing-an-n1-performance-problem-in-xaf-xpo-with-totally-undocumented-apis.md

```diff
@@ -0,0 +1,948 @@
+19.09.20 12:08:07.941 Result: rowcount = 55, total = 2228, N0.{Oid,Guid} = 880, N0.{Name,String} = 908, N0.{OptimisticLockField,Int32} = 220, N0.{GCRecord,Int32} = 220
+19.09.20 12:08:07.942 Executing sql 'select N0."Oid",N0."Hours",N0."SlowOfferWithPersistentAlias",N0."OptimisticLockField",N0."GCRecord" from "dbo"."SlowOfferItemWithPersistentAlias" N0 where (N0."GCRecord" is null and (N0."SlowOfferWithPersistentAlias" = @p0))' with parameters {46ba6afd-f256-4022-bdea-fd2b2f782897}
+19.09.20 12:08:07.960 Result: rowcount = 1000, total = 44000, N0.{Oid,Guid} = 16000, N0.{Hours,Int32} = 4000, N0.{SlowOfferWithPersistentAlias,Guid} = 16000, N0.{OptimisticLockField,Int32} = 4000, N0.{GCRecord,Int32} = 4000
+.
+.
+.
+19.09.20 12:08:08.704 Executing sql 'select N0."Oid",N0."Hours",N0."SlowOfferWithPersistentAlias",N0."OptimisticLockField",N0."GCRecord" from "dbo"."SlowOfferItemWithPersistentAlias" N0 where (N0."GCRecord" is null and (N0."SlowOfferWithPersistentAlias" = @p0))' with parameters {1a57b4dc-7812-48cc-83ae-15e184078c5f}
+19.09.20 12:08:08.718 Result: rowcount = 1000, total = 44000, N0.{Oid,Guid} = 16000, N0.{Hours,Int32} = 4000, N0.{SlowOfferWithPersistentAlias,Guid} = 16000, N0.{OptimisticLockField,Int32} = 4000, N0.{GCRecord,Int32} = 4000
+```
+
+This results in the following performance
+
+First user interaction:
+
+1. Sort by `Name`: 0.040 seconds
+2. Sort by `HourSum`: 0.030 seconds
+
+Visible data loaded:
+
+1. Sort by `Name`: ~~0.800 seconds
+2. Sort by `HourSum`: ~~0.800 seconds
+
+### InstantFeedbackView
+
+In `InstantFeedbackView` mode records are loaded in an async manor. It operates almost the same as the `DataView` mode in terms of number of queries, but the user experience **can be** much better, cause all records will be fetched in the background.
```

Comment:
```suggestion
In `InstantFeedbackView` mode records are loaded in an asynchronously. It operates almost the same as the `DataView` mode in terms of number of queries, but the user experience **can be** much better, cause all records will be fetched in the background.
```

Reply:
Shouldnt that be: loaded asynchronosly? Minus *in an*